src/num.jl

```diff
@@ -25,6 +25,8 @@ SymbolicUtils.@number_methods(
                               Num(f(value(a), value(b))),
                               [conj, real, transpose]
                              )
+
+NaNMath.pow(x::Num, y::Integer) = wrap(NaNMath.pow(unwrap(x), y))
```

Comment:
We dont't want that. It should just fall back to `^`.

Reply:
Why not? `@number_methods` above already defines `NaNMath.pow` this way

Reply:
Because NaNMath.pow returns floats for integer arguments in NaNMath < 1.0.2 which breaks eg Catalyst (see the issue in SymbolicUtils) and NaNMath >= 1.0.2 just falls back to `^` if the second argument is an `Integer`. So directly using `^` is the only way that works with all NaNMath versions and is what NaNMath does anyway in newer versions.

Reply:
Thanks for the clarification, I just saw your comments on the SymbolicUtils PR.